ignite/metrics/psnr.py

```diff
@@ -0,0 +1,115 @@
+from typing import Callable, Optional, Sequence, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["PSNR"]
+
+
+class PSNR(Metric):
+    r"""Computes average `Peak signal-to-noise ratio (PSNR) <https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio>`_.
+
+    .. math::
+        \text{PSNR}(i, j) = 10 * \log_{10}\left(\frac{ MAX_{i}^2 }{ \text{ MSE } }\right)
+
+    where :math:`\text{MSE}` is `mean squared error <https://en.wikipedia.org/wiki/Mean_squared_error>`_.
+
+    - `y_pred` and `y` must be in the following shape (batch_size, ...).
+    - `y_pred` and `y` must have same dtype and same shape.
+
+    Args:
+        data_range (int or float): Range of the image. If not provided, it will be determined
+            from the image type. Typically, ``1.0`` or ``255``.
+        output_transform (callable, optional): A callable that is used to transform the Engine’s
+            process_function’s output into the form expected by the metric.
+        device (str or torch.device): specifies which device updates are accumulated on.
+            Setting the metric’s device to be the same as your update arguments ensures
+            the update method is non-blocking. By default, CPU.
+
+    Example:
+
+    To use with ``Engine`` and ``process_function``, simply attach the metric instance to the engine.
+    The output of the engine's ``process_function`` needs to be in format of
+    ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y, ...}``.
+
+    .. code-block:: python
+
+        def process_function(engine, batch):
+            # ...
+            return y_pred, y
+        engine = Engine(process_function)
+        psnr = PSNR()
+        psnr.attach(engine, "psnr")
```

Comment:
```suggestion
        psnr.attach(engine, "psnr")
    # ...
    state = engine.run(data)
    print(f"PSNR: {state.metrics['psnr']}")
```
I was thinking that providing a complete example could be more helpful. Same in this PR : https://github.com/pytorch/ignite/pull/1583/files